server/app/controllers/dev/DevToolsController.java

```diff
@@ -133,6 +133,12 @@ public Result seedProgramsHeadless() {
     return seedProgramsInternal() ? ok() : internalServerError();
   }
 
+  public Result seedApplicationsHeadless(Request request) {
+    String programSlug = request.body().asFormUrlEncoded().get("programSlug")[0];
+    int count = Integer.parseInt(request.body().asFormUrlEncoded().get("count")[0]);
```

Comment:
(blocking) This could throw if there is not programSlug or it is null when accessing the index. Use Play's FormFactory (`formFactory.form().bindFromRequest`) here instead. It has more safeguards.